pkg/controller/deployment/deployment_controller.go

```diff
@@ -450,6 +451,11 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
+	// Cleanup old rcs
+	if deployment.Spec.RevisionHistoryLimit != nil {
```

Comment:
Hm, we would need to make an additional list for the RCs vs not having to reimplement it for every strategy. Any thoughts here? @ironcladlou